test_data/makedata_suites/402_views.js

```diff
@@ -0,0 +1,470 @@
+    storeValues: "none",
+    trackListPositions: false,
+    fields: {
+      animal: {
+        "cache": false,
+      },
+      name: {}
+    },
+    "analyzers": ["AqlAnalyzerHash"]
+  };
+
+  let links = [
+    link_no_cache,
+    link_cache_true_top,
+    link_cache_false_top,
+    link_cache_true_bottom,
+    link_cache_false_bottom,
+    link_cache_true_top_true_bottom,
+    link_cache_true_top_false_bottom,
+    link_cache_false_top_true_bottom,
+    link_cache_false_top_false_bottom
+  ];
+
+  let simulateNormalization = function (linkDefinition) {
+    // This function will simulate field normalization inside link definition.
+    /*
+    5 possible cases when we should omit 'cache' value from link definition:
+          1) ____                2)  ___
+                 ____                  'cache': false
+  
+  
+          3)'cache': false    4) 'cache': false     5) 'cache': true
+                  ____               'cache': false         'cache': true
+    */
+
+    let result = linkDefinition;
+    // remove 'cache' values from link definition
+    if (result.hasOwnProperty("cache")) {
+      if (result["cache"] == false) {
+
+        if (result["fields"]["animal"].hasOwnProperty("cache")) {
+
+          if (result["fields"]["animal"]["cache"] == false) {
+
+            delete result["cache"];
+            delete result["fields"]["animal"]["cache"];
+          } else {
+            delete result["cache"];
+          }
+        } else {
+          delete result["cache"];
+        }
+      } else {
+        if (result["fields"]["animal"].hasOwnProperty("cache")) {
+          if (result["fields"]["animal"]["cache"] == true) {
+            delete result["fields"]["animal"]["cache"];
+          }
+        }
+      }
+    } else {
+
+      if (result["fields"]["animal"].hasOwnProperty("cache")) {
+
+        if (result["fields"]["animal"]["cache"] == false) {
+
+          delete result["fields"]["animal"]["cache"];
+        }
+      }
+    }
+
+    return result;
+  };
+
+  let removeCacheFields = function (linkDefinition) {
+    // This function will simulate field normalization when 'cache' field is not supported
+    // i.e. it will be simply ommited everywhere
+
+    let result = linkDefinition;
+    if (result.hasOwnProperty("cache")) {
+      delete result["cache"];
+    }
+    if (result["fields"]["animal"].hasOwnProperty("cache")) {
+      delete result["fields"]["animal"]["cache"];
+    }
+
+    return result;
+  };
+
+  let compareLinks = function (cacheSizeSupported, linkFromView, expectedRawLink) {
+
+    let expectedLink;
+    if (cacheSizeSupported) {
+      expectedLink = simulateNormalization(expectedRawLink);
+    } else {
+      expectedLink = removeCacheFields(expectedRawLink);
+    }
+
+    // remove redundant 'utilizeCache' values. 
+    delete expectedLink["utilizeCache"];
+
+    // actual comparison
+    return _.isEqual(linkFromView, expectedLink);
+  };
+
+  let getMetricValue = function (text, name) {
```

Comment:
DRY pls. Move these functions to ../common.py, and adjust 950 to use these as well.